packages/vite/bin/vite.js

```diff
@@ -7,6 +7,10 @@ if (!import.meta.url.includes('node_modules')) {
     // only available as dev dependency
     await import('source-map-support').then((r) => r.default.install())
   } catch {}
+
+  process.on('unhandledRejection', (err) => {
+    throw new Error('UNHANDLED PROMISE REJECTION', { cause: err })
+  })
```

Comment:
This won't catch the reported case. But will catch a similar issue in some cases.

Reply:
Is it useful to add this by default in the CLI? Shouldn't we preserve the unhandled rejections error as is? Re-throwing the error seems to only wrap it around

Reply:
https://github.com/vitejs/vite/pull/18886/files/1cc739eb0cd1be50d11b934a7502d8f646fca4de#diff-d41df705e208c4e8871565005d95b6fd86ba6eb17c826a225715203ba0c31a28R5
This only happens when developing Vite core or linking to local, so I think it's fine.

> Re-throwing the error seems to only wrap it around

If I didn't test it wrongly, when an error is thrown inside a event handler, Node exits the process.

Reply:
> This only happens when developing Vite core or linking to local, so I think it's fine.

Ah ok I didn't notice that. 

> If I didn't test it wrongly, when an error is thrown inside a event handler, Node exits the process.

Ah so when developing locally we want to forcefully exit the process, mostly for debugging I guess? I think overall that makes sense to me then. Thanks

Reply:
Yep!